2022/Q4/artifacts/PoS-pseudocode/PoS-model.md

```diff
@@ -289,7 +289,13 @@ func unbond(validator_address, delegator_address, total_amount)
 
         validators[validator_address].unbond_records[cur_epoch+pipeline_length][start] += amount_unbonded
         remain -= amount_unbonded
-      update_total_deltas(validator_address, pipeline_length, -1*amount_after_slashing)
+      
+      // Ensure that the validator's stake does not go negative due to the slashing
+      var pipeline_stake = read_epoched_field(validators[validator_address].total_deltas, cur_epoch + pipeline_length, 0)
+      var token_change = min{ amount_after_slashing, pipeline_stake }
```

Comment:
I think you may be right in thinking this may not be necessary. It's possible I threw it in there when I was initially trying to prevent negative stake in lots of places. I've just tested the removal of this logic with our native state machine test and the test passed.

I'll remove this for now and remove it from the Namada code in a new PR.